### DISCUSSION TITLE
How can need it configurate NPM for guacamole apache

### DISCUSSION THREAD

**Discussion by nkgiovannivl** [NginxProxyManager/nginx-proxy-manager#2401] [Q&A]
Hi All,

Can you help me with configuration instructions for my NPM / guacamole, I'm using NPD with Home Assistant without any issue, but i can not find how need i to configure for guacamole, below my actual configuration:

![image](https://user-images.githubusercontent.com/48342344/202043186-202392f4-ce2a-44c9-b790-89450373c680.png)

![image](https://user-images.githubusercontent.com/48342344/202043253-66ea6493-42c1-4076-a59e-e07b67db0b5e.png)

![image](https://user-images.githubusercontent.com/48342344/202043312-7e07bfd9-21ca-4119-a4bf-0c99b408ada7.png)

Thanks for your support!

**Comment by luckyluk83** [comment]
location / {
    proxy_pass http://192.168.10.50:8080/guacamole/;
    proxy_buffering off;
    proxy_http_version 1.1;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection $http_connection;
    proxy_cookie_path /guacamole/ /;
    access_log off;
}



that is what I've have in advanced section and it working like that for months now

**Reply by nkgiovannivl** [reply]
> location / { proxy_pass http://192.168.10.50:8080/guacamole/; proxy_buffering off; proxy_http_version 1.1; proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for; proxy_set_header Upgrade $http_upgrade; proxy_set_header Connection $http_connection; proxy_cookie_path /guacamole/ /; access_log off; }
> 
> that is what I've have in advanced section and it working like that for months now

Thanks! it's works!

**Reply by luckyluk83** [reply]
You're welcome. Happy to help

**Reply by vulcanjedi** [reply]
> location / { proxy_pass http://192.168.10.50:8080/guacamole/; proxy_buffering off; proxy_http_version 1.1; proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for; proxy_set_header Upgrade $http_upgrade; proxy_set_header Connection $http_connection; proxy_cookie_path /guacamole/ /; access_log off; }
> 
> that is what I've have in advanced section and it working like that for months now

Ive done this but hasn't improved.  NPM the advanced tab says set/add headers doesnt work and need custom location? Any help plz?

**Reply by nkgiovannivl** [reply]
> > location / { proxy_pass http://192.168.10.50:8080/guacamole/; proxy_buffering off; proxy_http_version 1.1; proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for; proxy_set_header Upgrade $http_upgrade; proxy_set_header Connection $http_connection; proxy_cookie_path /guacamole/ /; access_log off; }
> > that is what I've have in advanced section and it working like that for months now
> 
> Ive done this but hasn't improved. NPM the advanced tab says set/add headers doesnt work and need custom location? Any help plz?

No custom location:
![image](https://user-images.githubusercontent.com/48342344/225491529-e34c95b4-549d-4e54-b4bc-72226068358f.png)

Just in advance:
![image](https://user-images.githubusercontent.com/48342344/225491591-33ad8b57-a976-4d05-8d62-a5ec34d6cdb1.png)

location / {
    proxy_pass http://192.168.1.5:8080/guacamole/;
    proxy_buffering off;
    proxy_http_version 1.1;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection $http_connection;
    proxy_cookie_path /guacamole/ /;
    access_log off;
}

**Comment by Darkmagister** [comment]
i'm having a similar issue, i've the same config but i'm still not able to connect beacuse i get a 403 in the /api/tokens request any idea why ? (guacamole page load but instead of the login page i get an error message)
